src/client_networking.rs

```diff
@@ -55,27 +55,47 @@ impl From<StrictError> for ConnectionError {
 }
 
 
-
+// I'd change the declaration to: request_table(table_name: &str, server_address: &str)
 pub fn request_csv(name: &str, address: &str) -> Result<StrictTable, ConnectionError> {
+    // Validating the server_address somehow and providing a descriptive error would probably be a good idea
 
     let mut connection: TcpStream = match TcpStream::connect(address) {
         Ok(stream) => stream,
         Err(e) => {return Err(ConnectionError::Io(e));}
     };
     
+    // The writing to the connection here is the same code as elsewhere
+    // perhaps you want to encapsulate the writing/communications themselves somehow. 
+    // E.g. say you change the server to accept json or another format, 
+    // you'd only want to make the formatting change in one place.
+    // It doesn't have to be a class but some intermideary between this request_table function
+    // and the connection that would surface say at least two methods: 
+    // send_request(method: requesting|sending|other, params: &str), 
+    //      where the method param would be an enum allowing you to 
+    //      switch between the appearent methods however that's best done in rust
+    //      and the other would be the params (like name) currently going into the formatted string.
+    //      This method could then also handle parsing the data from the server  (i.e. the read in the loop below)
+    //      and casting the bytes to a string
+    // and 
+    // send_confirmation(), which would send the "OK"
+
     match connection.write(format!("admin|admin|Requesting|{}", name).as_bytes()) {
         Ok(n) => println!("Wrote request as {n} bytes"),
         Err(e) => {return Err(ConnectionError::Io(e));},
     };
 
     let mut buffer: [u8; BUFFER_SIZE] = [0; BUFFER_SIZE];
+    // What the hell are you looping over here? Hahah, rust looks weird coming from js 😂 but very cool
```

Comment:
This loop is unnecessary here. But in rust loop{} just loops until there is a break